### ISSUE TITLE
Custom Babel Presets causing Error: Cannot find module 'babel-preset-'

### ISSUE THREAD

**Issue by MTCoster** [bdkjones/CodeKit#375]
**Quick, short summary:**
As per the title, I'm attempting to add a custom preset module to the Babel transpiler.  This results in the following error message:

```
Babel: Transpiling failed:

Error: Cannot find module 'babel-preset-'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at /Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:303:19
    at Array.map (native)
    at OptionManager.resolvePresets (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:281:20)
    at OptionManager.mergePresets (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:270:10)
    at OptionManager.mergeOptions (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:255:14)
    at OptionManager.init (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:385:12)

```

I've tried using values both with and without the `babel-preset-` prefix.  All modules have been installed globally with NPM.

**Expected results:**
A successful build.

**Actual results:**
The above error.

**Exact steps to reproduce:**
- Enable Babel transpilation for JavaScript files
- Add a custom Babel preset
- Attempt to process a JavaScript file 

**A link to download a simplified project or file that shows the issue:**
https://github.com/MTCoster/CK-Babel-Bug

**Your configuration (any details about your system that you think might be relevant)**

**Comment by treetrum** [comment]
I think this is probably related to the same issue as what I've been running into with #366 - which in turn is the same issue as #200.

The work around I use comes from @robstown:

> I figured it out. You get this error whenever you have 'react' & 'react-dom' loaded in your local node_modules folder and you select the 'react' babel preset option in CK3. These have nothing to do with babel, they are just the react npm bundles. The error goes away when I have all of the babel presets and options in my local node_modules folder instead of using the ones in the CK3 folder. I have no idea why that is? There must be a name conflict somewhere and it's picking out the wrong file?

**Comment by bdkjones** [comment]
Excellent report. 

Found the issue causing this and have fixed it for 3.3.

**Comment by bdkjones** [comment]
Note: although I modify the version of Babel in CodeKit to search for globally-installed presets and plugins, you should really consider installing those locally in your project instead. Babel (and JS transpiling in general) is a complicated nightmare and you'll frequently want to lock a project to a version of the plugins/presets that you KNOW worked for that project.
